ts/features/walletV3/history/store/reducers/index.ts

```diff
@@ -0,0 +1,144 @@
+import { getLookUpId } from "../../../../../utils/pmLookUpId";
+import {
+  walletPaymentCreateTransaction,
+  walletPaymentGetDetails,
+  walletPaymentGetTransactionInfo
+} from "../../../payment/store/actions/networking";
+import { walletPaymentInitState } from "../../../payment/store/actions/orchestration";
+import { WalletPaymentFailure } from "../../../payment/types/WalletPaymentFailure";
+import { PaymentHistory } from "../../types";
+import { walletPaymentHistoryStoreOutcome } from "../actions";
+
+export type WalletPaymentHistoryState = {
+  ongoingPayment?: PaymentHistory;
+  archive: ReadonlyArray<PaymentHistory>;
+};
+
+const INITIAL_STATE: WalletPaymentHistoryState = {
+  archive: []
+};
+
+export const ARCHIVE_SIZE = 15;
+
+const reducer = (
+  state: WalletPaymentHistoryState = INITIAL_STATE,
+  action: Action
+): WalletPaymentHistoryState => {
+  switch (action.type) {
+    case getType(walletPaymentInitState):
+      return {
+        ...state,
+        ongoingPayment: {
+          startOrigin: action.payload.startOrigin,
+          startedAt: new Date(),
+          lookupId: getLookUpId()
+        }
+      };
+    case getType(walletPaymentGetDetails.request):
+      return updatePaymentHistory(
+        state,
+        {
+          rptId: action.payload
+        },
+        true
+      );
+    case getType(walletPaymentGetDetails.success):
+      return updatePaymentHistory(state, {
+        verifiedData: action.payload
+      });
+    case getType(walletPaymentCreateTransaction.success):
+    case getType(walletPaymentGetTransactionInfo.success):
+      return updatePaymentHistory(state, {
+        transaction: action.payload
+      });
+    case getType(walletPaymentHistoryStoreOutcome):
+      return updatePaymentHistory(state, {
+        outcome: action.payload
+      });
+    case getType(walletPaymentGetDetails.failure):
+    case getType(walletPaymentCreateTransaction.failure):
+    case getType(walletPaymentGetTransactionInfo.failure):
+      return updatePaymentHistory(state, {
+        failure: pipe(
+          WalletPaymentFailure.decode(action.payload),
+          O.fromEither,
+          O.toUndefined
+        )
+      });
+    case getType(differentProfileLoggedIn):
+    case getType(clearCache):
+      return INITIAL_STATE;
+  }
+  return state;
+};
+
+const appendItemToArchive = (
+  archive: ReadonlyArray<PaymentHistory>,
+  item: PaymentHistory
+): ReadonlyArray<PaymentHistory> =>
+  pipe(
+    archive,
+    // Remove previous entry if already exists
+    a => a.filter(({ rptId }) => !_.isEqual(rptId, item.rptId)),
+    // Keep only the latest ARCHIVE_SIZE - 1 entries
+    a => a.slice(-ARCHIVE_SIZE + 1),
+    // Add the new entry to the archive
+    a => [...a, item]
+  );
+
+const replaceLastItemInArchive = (
+  archive: ReadonlyArray<PaymentHistory>,
+  item: PaymentHistory
+): ReadonlyArray<PaymentHistory> =>
+  pipe(
+    archive,
+    // Remove last element in archive
+    a => a.slice(0, a.length - 1),
```

Comment:
```suggestion
    _.dropRight,
```
this one though seems like a cleanup to me, LMK what you think about it

Reply:
Nice suggestion! See the changes here https://github.com/pagopa/io-app/pull/5515/commits/80c4771fe26df1a9c3394400aa7019fbaf17ad92
I've also cleaned up the function a bit